### DISCUSSION TITLE
Autostart on system login

### DISCUSSION THREAD

**Discussion by manueldeprada** [bpozdena/OneDriveGUI#89] [Q&A]
I did not see it in the documentation...

Is there an easy or recommended way to autostart OneDriveGUI after login?

Excellent work, thanks for this application!!

**Comment by bpozdena** [comment]
This will depend on your distribution and desktop environment. Most distros come with some kind of autostart application where you can add the AppImage. In some distros you need to make a link to to the AppImage into the user autostart directory.

Just search how to autostart apps in your distro.

**Reply by roke-julian-lockhart** [reply]
> Most distros come with some kind of autostart application where you can add the AppImage.

@bpozdena, not Fedora or openSUSE: AppImageKit provides this capability, but that's not included by default, and isn't per se reliable. Activating this per DE is more complex, too:

DE | Documentation
-|-
KDE Plasma | [`userbase.kde.org/index.php?title=System_Settings/Autostart&oldid=464627`](https://userbase.kde.org/index.php?title=System_Settings/Autostart&oldid=464627#:~:text=More%20Information-,System%20Settings/Autostart,-Page)
GNOME Desktop | [`help.gnome.org/users/gnome-help/stable/shell-apps-auto-start.html.en`](https://help.gnome.org/users/gnome-help/stable/shell-apps-auto-start.html.en#:~:text=Have%20applications%20start%20automatically%20on%20log%20in) [^1]

[^1]: Visit [`gitlab.gnome.org/GNOME/gnome-user-docs/-/blob/c53e11b7f869bc53d4f5c5ebfc3298aa8ad78e9d/gnome-help/C/shell-apps-auto-start.page#L26-70`](https://gitlab.gnome.org/GNOME/gnome-user-docs/-/blob/c53e11b7f869bc53d4f5c5ebfc3298aa8ad78e9d/gnome-help/C/shell-apps-auto-start.page#L26-70:~:text=Have%20applications%20start%20automatically%20on%20log%20in) if it's unavailable.

**Comment by RobinVdBroeck** [comment]
I do have a gist for a systemd service that you might find useful: https://gist.github.com/RobinVdBroeck/1bbbf684ad10a1125f358fdb039c8789

**Reply by roke-julian-lockhart** [reply]
This is the most cross-platform solution that's reliable.

**Comment by RedSnt** [comment]
Personally, I just make a .desktop file in ~/.config/autostart that points to the appimage of OneDriveGUI. In OneDriveGUI there's an option called "Start OneDriveGUI minimized", that when enabled makes it so OneDriveGUI starts in the notification tray silently.

**Reply by bpozdena** [reply]
The option to enable GUI autostart after login is now implemented in v1.3.0.
